ligo/src/registryDAO.mligo

```diff
@@ -120,29 +160,27 @@ let registry_DAO_decision_lambda (proposal, extras : proposal * contract_extra)
     proposal_metadata = proposal.metadata
     } in
   let proposal_key = to_proposal_key (propose_param, proposal.proposer) in
-  match Map.find_opt "agoraPostID" proposal.metadata with
-  | Some (packed_b) ->
+  match extract_proposal(proposal.metadata) with
+  | Normal_proposal diff ->
+      let ce_after_registry_update = apply_diff(diff, extras) in
+      let ce_after_registry_affected_update =
+        apply_diff_affected(proposal_key, diff, ce_after_registry_update) in
+        (([] : operation list), ce_after_registry_affected_update)
+  | Update_receivers_proposal urp ->
+      let current_set = require_unpacked_proposal_receivers(require_extra_value("proposal_receivers", extras)) in
       begin
-        match ((Bytes.unpack packed_b) : (nat option)) with
-        | Some (agora_post_id) ->
-            begin
-              match Map.find_opt "updates" proposal.metadata with
-              | Some (packed_b) ->
-                  begin
-                    match ((Bytes.unpack packed_b) : (registry_diff option)) with
-                    | Some (diff) ->
-                        let ce_after_registry_update = apply_diff(diff, extras) in
-                        let ce_after_registry_affected_update =
-                          apply_diff_affected(proposal_key, diff, ce_after_registry_update) in
-                          (([] : operation list), ce_after_registry_affected_update)
-                    | None ->
-                      (failwith "UPDATES_UNPACKING_FAILED" : operation list * contract_extra)
-                  end
-              | None -> (failwith "UPDATES_NOT_FOUND" : operation list * contract_extra)
-            end
-        | None -> (failwith "UNPACKING FAILED" : operation list * contract_extra)
+        match urp with
+        | Add_receivers receivers ->
+            let
+              foldFn (existing, item : proposal_receivers * address) : proposal_receivers = Set.add item existing in
+            let new_set = List.fold foldFn receivers current_set in
+            (([] : operation list), Map.update "proposal_receivers" (Some (Bytes.pack new_set)) extras)
+        | Remove_receivers receivers ->
+            let foldFn (existing, item : address set * address) : address set = Set.remove item existing in
+            let new_set = List.fold foldFn receivers current_set in
```

Comment:
Just to follow up on the comment above, I think it's fine here to distinguish the `new_set` from the `current_set`, just because we don't end up replacing the `current_set` _directly_, but use it in a different way (packing it in this case).